### ISSUE TITLE
Regresison: Borked wall (example file)

### ISSUE THREAD

**Issue by johanrd** [IfcOpenShell/IfcOpenShell#623]
2. Convert [623--wall--borked.ifc](https://github.com/IfcOpenShell/files/blob/master/618--wall--borked.ifc) with 12e9f7b941e0fca8d4f5ee97e5204c2c5c9eb30c (v0.6.0 June 15) and experience that the wall looks like this:
<img width="372" alt="Screenshot 2019-06-18 at 16 39 59" src="https://user-images.githubusercontent.com/4601554/59693815-cc259a00-91e7-11e9-8c94-1a22469f35b5.png">

2. Geometry from Solibri, SimpleBIM, DDS and previous versions of ifcopenshell (8d0c95a, 42e0058):
<img width="348" alt="Screenshot 2019-06-18 at 16 53 09" src="https://user-images.githubusercontent.com/4601554/59694896-997ca100-91e9-11e9-87b0-41606551987e.png">
<img width="300" alt="Screenshot 2019-06-18 at 16 51 20" src="https://user-images.githubusercontent.com/4601554/59694898-9bdefb00-91e9-11e9-97b3-0dea7e98f02a.png">

Caveat: I am not 100% sure what the expected geometry of this file is. Therefore IfcOpenShell may be correct, and all the others wrong. Nevertheless, I'm putting it up here for investigation.

**Comment by aothms** [comment]
At first I was happy, would this be an opening successfully subtracted in IfcOpenShell and not in other viewers, but it appears IfcOpenShell fails to subtract some halfspaces at the bottom of the geometry and doesn't realize.

Below you can see the DAE filesizes for the conversion results. So again it appears this unification is the culprit. I'll have to investigate how this has altered the operands.

~~~
sha|n|[623](https://github.com/IfcOpenShell/files/blob/master/623--wall--borked.ifc)
---|---|---
12e9f7b|1|0|7455
443b8c2|1|0|7455
826316c|1|0|7455
b0b844e|1|0|7455
f5b7d0d|1|0|7455
68068cb|1|0|7455
532f129|1|0|7455
1a57162|1|0|7455
901e805|1|0|10059
41972bb|1|0|10059
fd6de77|1|0|10059
1d5e560|1|0|10059
0fd3c69|1|0|10059
4792fd9|1|0|10059
587edca|1|0|10059
71118fb|1|0|10059
80bdb36|1|0|6686
013b7a7|1|0|6686
94bf798|1|0|6686
45136fc|1|0|6686
4822b57|1|0|6686
68151b9|1|0|6686
4df526f|1|0|6686
42e0058|1|0|6686
4b57a7b|1|0|6686
5ee32fb|1|0|6686
16cafb9|1|0|6686
~~~

**Comment by aothms** [comment]
Edit: No 6686 is the correct size. So this is again due to the triangulation of non-planar boundaries

**Comment by aothms** [comment]
Yes, this is due to collecting the right hand sides of the tree in one call, instead of iteratively walking down the tree. I have reported this to occt https://tracker.dev.opencascade.org/view.php?id=30793
